pkg/controller/knativeserving/common/images.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	servingv1alpha1 "github.com/knative/serving-operator/pkg/apis/serving/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// UpdateImage updates the image of the deployment with a new registry and tag
+// @param image takes the form of new-registry.io/full/path/${NAME}:new-tag
+func UpdateImage(deployment *appsv1.Deployment, registry *servingv1alpha1.Registry, log logr.Logger) {
+	image := imageName(deployment.GetName(), registry)
+	if image == "" {
+		return
+	}
+	image = strings.ReplaceAll(image, "${NAME}", deployment.GetName())
```

Comment:
Did you want `${NAME}` to work in `overrides` as well as `default`? If not, you may want the `${NAME}` substitution to happen in the `imageName` function (but see below).

Reply:
Done.